Doc/whatsnew/3.8.rst

```diff
@@ -242,6 +242,11 @@ Optimizations
   Note that this means that instances can no longer be weak-referenced and
   that arbitrary attributes can no longer be added to them.
 
+* The :class:`list` constructor does not overallocate the internal item buffer
+  if the input iterable has a known size (the input implements ``__len__``).
```

Comment:
nitpick: is it "size" or "length"? (__len__ means "length", no?)

Reply:
Yup, I used size as the CAPI call is `PyObject_Size`. Do you prefer to use `length `?

Reply:
Honestly, it's not important, so it's up to you :-) The doc says "length":
https://docs.python.org/dev/reference/datamodel.html#object.__len__

For me, "size" means more number of bytes, and "length" or "count" the number of items. But I'm used to the C language which is different :-)

Reply:
Let's use `length` as you suggest :)